File: HomeStock-backend/src/main/kotlin/com/snk/HomeStock/repository/model/Asset.kt
package com.snk.HomeStock.repository.model

import com.fasterxml.jackson.databind.JsonNode
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import org.hibernate.annotations.JdbcTypeCode
import org.hibernate.type.SqlTypes
import java.time.OffsetDateTime

@Entity
@Table(name = "asset")
class Asset(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    @Column(nullable = false, unique = true)
    var name: String,

    @JdbcTypeCode(SqlTypes.JSON)
    @Column(columnDefinition = "jsonb")
    var metadata: JsonNode? = null,

    @Column(name = "date_photo")
    var datePhoto: OffsetDateTime? = null,

    @Column(name = "date_upload", nullable = false)
    var dateUpload: OffsetDateTime,

    @Column
    var origin: String? = null
)

File: modules/photo-module/server/entities/Asset.ts
import { Entity, PrimaryGeneratedColumn, Column, CreateDateColumn } from "typeorm";

@Entity()
export class Asset {
    @PrimaryGeneratedColumn("identity", { generatedIdentity: "ALWAYS" })
    id!: number;

    @Column("text", { unique: true })
    name!: string;

    @Column("jsonb", { nullable: true })
    metadata!: Record<string, any>;

    @Column("timestamptz", { nullable: true })
    date_photo!: Date;

    @CreateDateColumn({ type: "timestamptz" })
    date_upload!: Date;

    @Column("text", { nullable: true })
    origin!: string;
}
